src/components/dashboard/SendLinkSummary.js

```diff
@@ -91,21 +89,20 @@ const SendLinkSummary = ({ screenProps, styles }: AmountProps) => {
       generateSendShareText,
       generateSendShareObject,
       amount,
-      reason,
       counterPartyDisplayName,
       profile,
       setShared,
       showDialog,
-      screenProps,
+      push,
     ]
   )
 
   // Going to root after shared
   useEffect(() => {
     if (shared) {
-      screenProps.goToRoot()
+      goToRoot()
     }
-  }, [shared])
+  }, [shared, goToRoot])
```

Comment:
you need to go to root after `shared` becomes true only
so  you don't need goToRoot in deps as effect will be ran if on goToRoot function reference changes too

```
useEffect(() => {
     if (shared) {
       goToRoot()
     }
   }, [shared])
```
also you could use one-liner:

```
() => shared && gotoRoot()
```
or
```
() => shared ? gotoRoot() : null
```

Reply:
each variable should be added to deps in useCallback / useMemo only
in useEffect you need to pass only the vars you actually need to track

Reply:
Opps. I missed that it is a useEffect and not useCallback)
Fixed in new commit.